datafusion/optimizer/src/optimizer.rs

```diff
@@ -89,12 +110,12 @@ pub struct OptimizerConfig {
     max_passes: u8,
 }
 
-impl OptimizerConfig {
+impl OptimizerContext {
     /// Create optimizer config
     pub fn new() -> Self {
         Self {
             query_execution_start_time: Utc::now(),
-            next_id: 0, // useful for generating things like unique subquery aliases
+            next_id: AtomicUsize::new(1),
```

Comment:
We use 1, as previously the implementation returned the value post-increment which the atomics don't provide